### PR TITLE
feat: remove 'include_private'

### PR DESCRIPTION
**Pull request by rickstaa**
As explained in #2517 the 'include_private' flag is confusing and doesn't work as expected. This commit therefore removes this flag.

# TODOs

- [x] remove conflicts.
- [x] Update [FAQ](https://github.com/anuraghazra/github-readme-stats/discussions/1770#how-to-count-private-stats).

### CODE REVIEW

**Comment by rickstaa** [comment]
@qwerty541 @anuraghazra told me he will add you as a collaborator when he finds the time. I was curious to know what you think about this change?

**Review by qwerty541 — COMMENTED** [review]
> @anuraghazra told me he will add you as a collaborator when he finds the time.

Nice

> I was curious to know what you think about this change?

I fully agree that the `count_private` parameter has behavior that confuse users and support the course to change how it works. But it is important to consider that a large number of users still want to be able to add their private contributions to the statistics card, and I am one of them. As far as I understand, this pull request is a preparation step for adding the optional ability to display the value of the total contributions, including private ones, on the statistics card. If I understood everything correctly, then I support this change.

I can also add that I think it's good practice for a pull request to contain changes related to only one small task. If I were you, I would format the readme file in a separate pull request so that changes are not confusing.

**Comment by rickstaa** [comment]
> > @anuraghazra told me he will add you as a collaborator when he finds the time.
> 
> Nice
> 
> > I was curious to know what you think about this change?
> 
> I fully agree that the `count_private` parameter has behavior that confuse users and support the course to change how it works. But it is important to consider that a large number of users still want to be able to add their private contributions to the statistics card, and I am one of them. As far as I understand, this pull request is a preparation step for adding the optional ability to display the value of the total contributions, including private ones, on the statistics card. If I understood everything correctly, then I support this change.
> 
> I can also add that I think it's good practice for a pull request to contain changes related to only one small task. If I were you, I would format the readme file in a separate pull request so that changes are not confusing.

@qwerty541 This pull request was meant to remove the `count_private` flag all together, as we don't have permissions to view a user's private commits. Users that want to include private commits therefore have to [deploy their own Vercel instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own). However, You are welcome to create a pull request that shows contributions instead of commits (see https://github.com/anuraghazra/github-readme-stats/issues/2517#issuecomment-1554195386) 👍🏻.

**Comment by francois-rozet** [comment]
My take is that we can either remove "count_private" altogether or display contributions instead of commits. Note that "contributed to" can also be replaced by "repositories" to avoid confusions (while still counting the repositories that were contributed to).

**Comment by rickstaa** [comment]
> I also think displaying contributions instead of commits would be better as some users mainly have private contribs. However this will generate a lot of questions from users. Note that "contributed to" can also be replaced by "repositories" to avoid confusions (while still counting the repositories that were contributed to).

@francois-rozet Thanks for your feedback 💪🏻!

Users can already show private commits by deploying their own Vercel instance. However, as stated above, I agree that giving users the ability to also show contributions using the `show_contribs` and `include_private_contribs` flags while removing the `include_private` flag will be great 🚀! If anybody wants to take this one, feel free to create a pull request.

**Comment by francois-rozet** [comment]
Would `show_contribs` replace the commit field or add a new field? Also, we should directly count the private contributions instead of adding a flag `include_private_contribs`. I don't think anyone would choose to no count them if they are set as public in their GitHub profile.

**Comment by rickstaa** [comment]
> Would `show_contribs` replace the commit field or add a new field? Also, we should directly count the private contributions instead of adding a flag `include_private_contribs`. I don't think anyone would choose to not count them if they are set as public in their GitHub profile.

Ideally, I think it would be an extra statistics people can enable. People can then disable the commits if they like, using the `hide` flag. It could be added in a `extra_stats` array together with, for example, https://github.com/anuraghazra/github-readme-stats/pull/1404 and the number of answered discussions. I agree that the `include_private_contribs` flag is not needed 😅!

**Comment by francois-rozet** [comment]
I am not familiar with that part of the API (adding a new field to the card), but I can do the "backend" part if needed.

**Comment by qwerty541** [comment]
> > > @anuraghazra told me he will add you as a collaborator when he finds the time.
> > 
> > 
> > Nice
> > > I was curious to know what you think about this change?
> > 
> > 
> > I fully agree that the `count_private` parameter has behavior that confuse users and support the course to change how it works. But it is important to consider that a large number of users still want to be able to add their private contributions to the statistics card, and I am one of them. As far as I understand, this pull request is a preparation step for adding the optional ability to display the value of the total contributions, including private ones, on the statistics card. If I understood everything correctly, then I support this change.
> > I can also add that I think it's good practice for a pull request to contain changes related to only one small task. If I were you, I would format the readme file in a separate pull request so that changes are not confusing.
> 
> @qwerty541 This pull request was meant to remove the `count_private` flag all together, as we don't have permissions to view a user's private commits. Users that want to include private commits therefore have to [deploy their own Vercel instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own). However, You are welcome to create a pull request that shows contributions instead of commits (see [#2517 (comment)](https://github.com/anuraghazra/github-readme-stats/issues/2517#issuecomment-1554195386)) 👍🏻.

I looked a little deeper and realized that the main problem with the `count_private` parameter is not that its behavior is not obvious to users, but that because of it, when using your own vercel instance, the statistics card shows false information.  When using your own vercel instance, the statistics card already shows all private contributions, and the `count_private` parameter adds redundant commits that are already taken into account there. Now I fully agree that this parameter should be removed.

**Comment by rickstaa** [comment]
> > > > @anuraghazra told me he will add you as a collaborator when he finds the time.
> > > 
> > > 
> > > Nice
> > > > I was curious to know what you think about this change?
> > > 
> > > 
> > > I fully agree that the `count_private` parameter has behavior that confuse users and support the course to change how it works. But it is important to consider that a large number of users still want to be able to add their private contributions to the statistics card, and I am one of them. As far as I understand, this pull request is a preparation step for adding the optional ability to display the value of the total contributions, including private ones, on the statistics card. If I understood everything correctly, then I support this change.
> > > I can also add that I think it's good practice for a pull request to contain changes related to only one small task. If I were you, I would format the readme file in a separate pull request so that changes are not confusing.
> > 
> > 
> > @qwerty541 This pull request was meant to remove the `count_private` flag all together, as we don't have permissions to view a user's private commits. Users that want to include private commits therefore have to [deploy their own Vercel instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own). However, You are welcome to create a pull request that shows contributions instead of commits (see [#2517 (comment)](https://github.com/anuraghazra/github-readme-stats/issues/2517#issuecomment-1554195386)) 👍🏻.
> 
> I looked a little deeper and realized that the main problem with the `count_private` parameter is not that its behavior is not obvious to users, but that because of it, when using your own vercel instance, the statistics card shows false information. When using your own vercel instance, the statistics card already shows all private contributions, and the `count_private` parameter adds redundant commits that are already taken into account there. Now I fully agree that this parameter should be removed.

Great! I will merge this pull request when I get a review from @anuraghazra or @Zo-Bro-23 👍🏻.

**Comment by qwerty541** [comment]
Hey, @rickstaa! You need to resolve conflicts before merging.

**Comment by rickstaa** [comment]
> Hey, @rickstaa! You need to resolve conflicts before merging.

Of course! Sorry, here you go 😄.

File: readme.md [hunk 1]
```diff
@@ -116,6 +116,10 @@ Change the `?username=` value to your GitHub username.
 > **Note**
 > Available ranks are S+ (top 1%), S (top 25%), A++ (top 45%), A+ (top 60%), and B+ (everyone). The values are calculated by using the [cumulative distribution function](https://en.wikipedia.org/wiki/Cumulative_distribution_function) using commits, contributions, issues, stars, pull requests, followers, and owned repositories. The implementation can be investigated at [src/calculateRank.js](./src/calculateRank.js).
 
+### Include statistics from private repositories
+
+We do not have permission to see your private repositories. Therefore, we cannot include statistics like stars, commits and pull requests for these private repositories on the public Vercel instance. If you also want to show these private statistics on the stats card, you must [deploy your own Vercel instance](#deploy-on-your-own).
+
```
Comment (qwerty541) [inline readme.md:122]: Looks like this change duplicates warning that i have added in #2776. I think that only one note about that should be left. You should merge them or remove any.

Comment (rickstaa) [reply readme.md]: Your right, I updated this pull request. The language private deploy note now also is a warning for consistency.

**Pull request merged**